esti/lakectl_local_test.go

```diff
@@ -406,3 +407,215 @@ func TestLakectlLocal_commit(t *testing.T) {
+			fileName := "test.txt"
+			fd, err = os.Create(fileName)
+			require.NoError(t, err)
+			require.NoError(t, fd.Truncate(1e9))
+			require.NoError(t, fd.Close())
+			runCmd(t, Lakectl()+" fs upload -s "+fileName+" lakefs://"+repoName+"/"+vars["BRANCH"]+vars["PREFIX"]+"/"+fileName, false, false, vars)
+			runCmd(t, Lakectl()+" commit lakefs://"+repoName+"/"+vars["BRANCH"]+" --allow-empty-message -m \" \"", false, false, vars)
+
+			// Pull changes and interrupt
+			RunCmdAndVerifyContainsTextWithTimeout(t, Lakectl()+" local pull "+dataDir, true, false, "", vars, time.Millisecond*100)
+
+			// Pull changes without force flag
+			runCmd(t, Lakectl()+" local pull "+dataDir, true, false, vars)
+
+			// Pull changes and verify data
+			runCmd(t, Lakectl()+" local pull "+dataDir+" --force", false, false, vars)
+			localVerifyDirContents(t, dataDir, []string{fileName})
+		})
+	}
+}
+
+func TestLakectlLocal_interruptedClone(t *testing.T) {
+	tmpDir := t.TempDir()
+	fd, err := os.CreateTemp(tmpDir, "")
+	require.NoError(t, err)
+	require.NoError(t, fd.Close())
+	dataDir, err := os.MkdirTemp(tmpDir, "")
+	require.NoError(t, err)
+	repoName := generateUniqueRepositoryName()
+	storage := generateUniqueStorageNamespace(repoName)
+	vars := map[string]string{
+		"REPO":    repoName,
+		"STORAGE": storage,
+		"BRANCH":  mainBranch,
+		"REF":     mainBranch,
+	}
+
+	// No repo
+	vars["LOCAL_DIR"] = tmpDir
+	RunCmdAndVerifyFailureWithFile(t, Lakectl()+" local clone lakefs://"+repoName+"/"+mainBranch+"/ "+tmpDir, false, "lakectl_local_clone_non_empty", vars)
+
+	runCmd(t, Lakectl()+" repo create lakefs://"+repoName+" "+storage, false, false, vars)
+	runCmd(t, Lakectl()+" log lakefs://"+repoName+"/"+mainBranch, false, false, vars)
+
+	// Bad ref
+	RunCmdAndVerifyFailureWithFile(t, Lakectl()+" local init lakefs://"+repoName+"/bad_ref/ "+tmpDir, false, "lakectl_local_commit_not_found", vars)
+
+	prefix := "images"
+	objects := []string{
+		"ro_1k.1",
+		"ro_1k.2",
+		"ro_1k.3",
+		prefix + "/1.png",
+		prefix + "/2.png",
+		prefix + "/3.png",
+		prefix + "/subdir/1.png",
+		prefix + "/subdir/2.png",
+		prefix + "/subdir/3.png",
+	}
+
+	localCreateTestData(t, vars, objects)
+
+	// Upload and commit a large object, so we will have a chance to interrupt the pull before it finishes
+	fileName := "test.txt"
+	fd, err = os.Create(fileName)
+	require.NoError(t, err)
+	require.NoError(t, fd.Truncate(1e9))
+	require.NoError(t, fd.Close())
+	runCmd(t, Lakectl()+" fs upload -s "+fileName+" lakefs://"+repoName+"/"+mainBranch+"/"+prefix+"/"+fileName, false, false, vars)
+	runCmd(t, Lakectl()+" commit lakefs://"+repoName+"/"+mainBranch+" --allow-empty-message -m \" \"", false, false, vars)
+
+	vars["LOCAL_DIR"] = dataDir
+	vars["PREFIX"] = "images"
+
+	// Clone changes and interrupt
+	RunCmdAndVerifyContainsTextWithTimeout(t, Lakectl()+" local clone lakefs://"+repoName+"/"+mainBranch+"/"+prefix+" "+dataDir, true, false, "", vars, time.Millisecond*100)
```

Comment:
Having a timeout here is also scary: It has to be short otherwise the file needs to be even bigger.  But what if it takes >100msec?   That can happen on a busy testing server.